packages/flutter_tools/test/general.shard/vmservice_test.dart

```diff
@@ -280,8 +281,127 @@ void main() {
       ]))
     ]));
   });
+
+  testWithoutContext('runInView forwards arguments correctly', () async {
+    final FakeVmServiceHost fakeVmServiceHost = FakeVmServiceHost(
```

Comment:
Unlike the FakeProcessManager, this has to support both request/response and notifications. The latter have no corresponding request (like a log message, or isolate notification).

Combining these two concepts as a `VmServiceExpectation` allows us to describe  a more complex operation like runInView, which we need to ack the streamListen, ack the runInView request, and then return an isolate runnable notification